### PR TITLE
node: Added overloads of ChildProcess#send() method

### PR DESCRIPTION
**Pull request by avchugaev**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://nodejs.org/api/child_process.html#child_process_child_send_message_sendhandle_options_callback>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

Added declarations of possible overloads of `ChildProcess#send()` method according to latest edit of documentation.

### CODE REVIEW

File: types/node/index.d.ts [hunk 1]
```diff
@@ -1782,6 +1782,9 @@ declare module "child_process" {
         stdio: [stream.Writable, stream.Readable, stream.Readable];
         pid: number;
         kill(signal?: string): void;
+        send(message: any): boolean;
```
Comment (Flarna) [inline types/node/index.d.ts]: I think this one is not needed as the next one with optional callback should cover the `send(message)` and `send(message, cb)` use cases.

File: types/node/index.d.ts [hunk 2]
```diff
@@ -1782,6 +1782,9 @@ declare module "child_process" {
...
+        send(message: any, callback?: (error: Error) => void): boolean;
+        send(message: any, sendHandle?: net.Socket | net.Server, options?: MessageOptions): boolean;
```
Comment (Flarna) [inline types/node/index.d.ts]: I think this one is not needed as the next one with optional callback after options covers the `send(message, sendHandle, options)` and `send(message, sendHandle, options, cb)` use cases.
But I think the use case `send(message, sendHandle, cb)` is not yet covered.

**Review by Flarna — CHANGES_REQUESTED** [review]
Please add some tests to cover all usecases allowed.

**Comment by achugaev93** [comment]
@Flarna I have updated signatures. You're right, missed case with 3 arguments. I looked at tests file, but i don't understand, how to write own. I familiar with Jest/Jasmine, but it seems there are another engine/approach.

**Comment by Flarna** [comment]
Tests are quite simple - it's just validated that `node-tests.ts` contains valid typescript.
There is already a section `namespace child_process_tests` which contains some use of child process, just add a few lines using the API you added/improved.

**Comment by achugaev93** [comment]
Changes in this PR - https://github.com/DefinitelyTyped/DefinitelyTyped/pull/17959

**Comment by RyanCavanaugh** [comment]
Closing per comments in #17959

**Pull request closed without merging**